raiden-ts/src/channels/epics.ts

```diff
@@ -677,67 +701,64 @@ export const channelDepositEpic = (
       );
       const tokenContract = getContractWithSigner(getTokenContract(token), onchainSigner);
       const tokenNetworkContract = getContractWithSigner(
-        getTokenNetworkContract(action.meta.tokenNetwork),
+        getTokenNetworkContract(tokenNetwork),
         onchainSigner,
       );
-      const channel: Channel = get(state.channels, [
-        action.meta.tokenNetwork,
-        action.meta.partner,
-      ]);
-      if (!channel || channel.state !== ChannelState.open || channel.id === undefined) {
+      const channel = state.channels[tokenNetwork][partner];
+      if (channel?.state !== ChannelState.open) {
         const error = new RaidenError(ErrorCodes.CNL_NO_OPEN_CHANNEL_FOUND, action.meta);
         return of(channelDeposit.failure(error, action.meta));
       }
-      const channelId = channel.id;
 
       // send approve transaction
-      return from(
-        tokenContract.functions.approve(action.meta.tokenNetwork, action.payload.deposit),
-      )
+      return from(tokenContract.functions.approve(tokenNetwork, action.payload.deposit))
         .pipe(
           tap(tx => log.debug(`sent approve tx "${tx.hash}" to "${token}"`)),
-          mergeMap(async tx => ({ receipt: await tx.wait(), tx })),
-          map(({ receipt, tx }) => {
-            if (!receipt.status)
-              throw new RaidenError(ErrorCodes.CNL_APPROVE_TRANSACTION_FAILED, {
-                token,
-                transactionHash: tx.hash!,
-              });
-            return tx.hash;
-          }),
-          tap(txHash => log.debug(`approve tx "${txHash}" successfuly mined!`)),
+          mergeMap(tx =>
+            from(tx.wait()).pipe(
+              map(receipt => {
+                if (!receipt.status)
+                  throw new RaidenError(ErrorCodes.CNL_APPROVE_TRANSACTION_FAILED, {
+                    token,
+                    transactionHash: tx.hash!,
+                  });
+                log.debug(`approve tx "${tx.hash}" successfuly mined!`);
+                return tx.hash;
+              }),
+            ),
+          ),
```

Comment:
We do this quite often:
* Wait for transaction
* Throw an error if status is invalid
* Otherwise return transaction hash

Might be good refactor candidate

Reply:
That is a really nice idea.

Reply:
Well spotted. I'll try. Thanks.

Reply:
Done, `assertTx`